src/features/element-hiding.js

```diff
@@ -343,6 +396,7 @@ export default class ElementHiding extends ContentFeature {
         setTimeout(() => {
             appliedRules = new Set()
             hiddenElements = new WeakMap()
-        }, 3100)
+            modifiedElements = new WeakMap()
+        }, clearCacheTimer)
```

Comment:
This was an oversight on my part, really should have been included in #682.

Reply:
I'm not sure this actually is guaranteed to finish after unhideTimeout -1.

I don't think it matters all that much but it might? We could resolve promises in the timers and await them all if we want to be sure.

Reply:
No you're right, it doesn't have to be exact - we just want it to be somewhat close to the upper bound of our checks. Let's say we end up clearing these at 12100ms:
- The last iteration of `collapseDomNode` happens with a 10000ms timeout. Seems unlikely that this would actually happen a full 2100ms later, but in the off chance it did, [this](https://github.com/duckduckgo/content-scope-scripts/blob/5cead52e6b95cfde349279d0e895b1106da31c50/src/features/element-hiding.js#L31) wouldn't return for an element that has already been hidden/modified, so we'd end up checking/hiding/modifying elements that had already been addressed. All of the timing based modifications we're making here are idempotent, so this wouldn't cause any issues.
- The last iteration of `expandNonEmptyDomNode` happens with a 12000ms timeout. If this were to occur after caches have been cleared, it likely wouldn't do anything, since `unhideNode` is only run if `alreadyHidden` is true, and `alreadyHidden` can only be true if `collapseDomNode` was run again after the caches were cleared. In the off chance `alreadyHidden` is actually true here, this would just check the hidden elements and unhide them if they contain visible content.